pkg/routes/common/pagination_test.go

```diff
@@ -46,3 +49,52 @@ func TestGetPagination(t *testing.T) {
 	}
 
 }
+func TestPaginate(t *testing.T) {
+	RegisterTestingT(t)
+	tt := []struct {
+		name     string
+		limit    int
+		offset   int
+		expected Pagination
+	}{
+		{name: "check limit value 10 offset 20", limit: 10, offset: 20, expected: Pagination{Limit: 10, Offset: 20}},
+		{name: "check limit value 10 offset 0", limit: 10, offset: 0, expected: Pagination{Limit: 10, Offset: defaultOffset}},
+		{name: "check limit value 0 offset 0", limit: 0, offset: 0, expected: Pagination{Limit: defaultLimit, Offset: defaultOffset}},
+		{name: "check limit value 10 offset 20", limit: 0, offset: 20, expected: Pagination{Limit: defaultLimit, Offset: 20}},
+	}
+
+	for _, te := range tt {
+		t.Run(te.name, func(t *testing.T) {
+			router := chi.NewRouter()
+			router.Use(Paginate)
+			router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+				pagination, ok := r.Context().Value(PaginationKey).(Pagination)
+				Expect(ok).To(BeTrue())
+				Expect(pagination.Limit).To(Equal(te.expected.Limit))
+				Expect(pagination.Offset).To(Equal(te.expected.Offset))
+			})
+			path := ""
+			if te.limit > 0 {
+				path = fmt.Sprintf("limit=%d", te.limit)
+			}
+			if te.offset > 0 {
+				offset := fmt.Sprintf("offset=%d", te.offset)
+				if path == "" {
+					path = offset
+				} else {
+					path = path + " " + offset
```

Comment:
very strange that this is working 
```suggestion
					path = path + "&" + offset
```